app/controllers/page_views_controller.rb

```diff
@@ -9,9 +9,10 @@ def create
                                 page_view_params.merge(counts_for_number_of_views: 10)
                               end
 
-    PageView.create(page_view_create_params)
+    PageView.create(page_view_create_params.
+      merge(visited_page_path: request.referer ? URI(request.referer).path : nil, visited_page_full_url: request.referer))
```

Comment:
there's a lot going on here that can be split in 2 or 3 lines: choosing the path, merging the params, creating the page view